libpod/runtime_pod_linux.go

```diff
@@ -300,6 +281,12 @@ func (r *Runtime) removePod(ctx context.Context, p *Pod, removeCtrs, force bool,
 		}
 	}
 
+	// let's unlock the containers so the cleanup processes can terminate their execution
+	for _, ctr := range ctrs {
```

Comment:
Just a small concern here, we are unlocking `containers` before remove is completed so other `podman` process have permission to modify containers and lock containers in a `pod` while the actual `pod` is being removed from another process.

However I'm unable to think of issues where this race could impact anybody.